ngx_http_access_token_to_jwt.c

```diff
@@ -414,8 +436,41 @@ static char *ngx_http_access_token_to_jwt_merge_loc_conf(ngx_conf_t *config, voi
     ngx_http_access_token_to_jwt_conf_t *prev = parent, *conf = child;
 
     ngx_conf_merge_value(conf->enable, prev->enable, 0);
-    ngx_conf_merge_str_value(conf->base64encoded_client_credentials, prev->base64encoded_client_credentials, "");
+
+    ngx_conf_merge_str_value(conf->client_id, prev->client_id, "");
+    ngx_conf_merge_str_value(conf->client_secret, prev->client_secret, "");
     ngx_conf_merge_str_value(conf->introspection_endpoint, prev->introspection_endpoint, "");
 
+
+    ngx_str_t  *concat_credentials =  ngx_pcalloc(config->pool, sizeof(ngx_str_t));
+
+    if (concat_credentials == NULL)
+    {
+        return NGX_CONF_ERROR;
+    }
+
+    int concat_credentials_size = conf->client_id.len + conf->client_secret.len + 1; // client_id:client_secret
+
+    u_char *concat_credentials_data = ngx_pcalloc(config->pool, concat_credentials_size);
+
+    if (concat_credentials_data == NULL)
+    {
+        return NGX_CONF_ERROR;
+    }
+
+    concat_credentials->data = concat_credentials_data;
+    concat_credentials->len = concat_credentials_size;
+
+    ngx_snprintf(concat_credentials_data, concat_credentials_size, "%s:%s", conf->client_id.data, conf->client_secret.data);
```

Comment:
ah, ` %V — ngx_str_t * ` It's a pointer, so you'd have to change to this:
```c
ngx_snprintf(..., "%V:%V", &conf->client_id, &conf->client_secret);
```
Sorry. learning here.